### PR TITLE
Improve E2E infra and adapt to VSTS

### PR DESCRIPTION
**Pull request by rohit21agrawal**
This PR does some changes in our E2E test infrastructure so it can run better on VSTS. Here is all that it does:

1) The currently running test name is now logged (instead of the ID earlier).
2) Corrects termination conditions for tests when they time out.
3) Copies the full PMC log to build drop folder along with test results.
4) Generates a JUnit XML test result report that VSTS can understand.
5) Fixes the count of tests - by including test cases in the number of tests.
6) Re-enables UWP test and UwpLegacyPackageRef tests.

I am currently working on adding support so that a screenshot is taken and the activity log recorded if a test times out, making it easier to investigate.

### CODE REVIEW

**Review by emgarten — APPROVED** [review]

File: test/EndToEnd/tests/ServicesTest.ps1 [hunk 1]
```diff
@@ -12,40 +12,43 @@
...
+function Test-MigrateVanilaUwpProjectJsonToPackageReference {
+    param(
+        $context
+    )
+
+    # Arrange
+    $p = New-UwpClassLibrary UwpClassLibrary1
+    $cm = Get-VsComponentModel
+    $projectDir = Get-ProjectDir $p
+    $result = [API.Test.InternalAPITestHook]::MigrateJsonProject($p.FullName)
```
Comment (jainaashish) [inline test/EndToEnd/tests/ServicesTest.ps1]: You may also want to check if project.json was exists when uwp project was created so you know it converted pj to pr

File: test/EndToEnd/NuGet.Tests.psm1 [hunk 2]
```diff
@@ -261,9 +282,19 @@ function Run-Test {
                         $name += "(" + [system.string]::join("_", $noteProperties) + ")"
                     }
                     $testCaseIndex++
-                    "Running Test case $name... ($testCaseIndex / $($testCases.Count))"
+                    $testIndexToPrint = $testIndex + $testCaseIndex
+                    "Running Test case $name... ($testIndexToPrint / $numberOfTests)"
```
Comment (jainaashish) [inline test/EndToEnd/NuGet.Tests.psm1:285]: Nice

**Review by jainaashish — APPROVED** [review]
LGTM

File: test/EndToEnd/tests/ServicesTest.ps1 (hunk 1 above)
Comment (rohit21agrawal) [reply test/EndToEnd/tests/ServicesTest.ps1]: the template we use has project.json so no need to check it again.

File: test/EndToEnd/NuGet.Tests.psm1 [hunk 3]
```diff
@@ -220,6 +243,7 @@ function Run-Test {
 
             $testCases = @( $null )
             $testCasesInfoString = ".`tThere are not multiple test cases. Just the 1 test"
+
```
Comment (nkolev92) [inline test/EndToEnd/NuGet.Tests.psm1]: nit: extra line.

**Review by nkolev92 — APPROVED** [review]
👑 but you don't get it until you fix the nit :P

**Pull request merged**